src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/resource/ManageIncentiveReviewsResource.kt

```diff
@@ -0,0 +1,253 @@
+        description = "Incentive review level information returned per prisoner",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect data specified to return incentive review Level History",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to use this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  suspend fun getCurrentIncentiveLevelForPrisoner(
+    @ArraySchema(schema = Schema(description = "List of booking Ids", required = true, type = "array"), arraySchema = Schema(type = "integer", format = "int64", pattern = "^[0-9]{1,20}$", additionalProperties = Schema.AdditionalPropertiesValue.FALSE))
+    @RequestBody
+    bookingIds: List<Long>,
+  ): List<CurrentIncentiveLevel> {
+    ensure {
+      ("bookingIds" to bookingIds).isNotEmpty()
+    }
+    return prisonerIncentiveReviewService.getCurrentIncentiveLevelForPrisoners(bookingIds)
+  }
+
+  @GetMapping("/prisoner/{prisonerNumber}")
+  @Operation(
+    summary = "Returns a history of incentive reviews for a prisoner",
+    description = "Prisoner Number is an unique reference for a prisoner in NOMIS",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Incentive review history information returned",
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect data specified to return incentive history",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to use this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))],
+      ),
+    ],
+  )
+  suspend fun getPrisonerIncentiveReviewHistory(
+    @Schema(description = "Prisoner Number", example = "A1234AB", required = true, pattern = "^[A-Z0-9]{7}$")
+    @PathVariable
+    prisonerNumber: String,
+  ): IncentiveReviewSummary =
+    prisonerIncentiveReviewService.getPrisonerIncentiveHistory(prisonerNumber)
+
+  @PostMapping("/booking/{bookingId}")
+  @PreAuthorize("hasRole('MAINTAIN_IEP') and hasAuthority('SCOPE_write')")
```

Comment:
IIRC this is the role users need to have right? But if these endpoint are meant to be used by system clients maybe we could introduce a separate role for this?

Maybe like `WRITE_INCENTIVES` or `WRITE_INCENTIVE_REVIEWS` similar to what we have in NONNA-API?

Also, you think the read role should be just called `READ_INCENTIVES` or `READ_INCENTIVE_REVIEWS`?
I'm probably in favour of being specific and call these `READ_INCENTIVE_REVIEWS`/`WRITE_INCENTIVE_REVIEWS` as it's more clear (especially as we shouldn't have any constraints on these names unless I'm missing something here)

Reply:
we could use scope instead for read/write 
- INCENTIVE_REVIEWS with read and write scope

Reply:
We could, it kinda makes sense but I don't know if there is any best practice there or common way of doing this across the other APIs.
I know in NONNA-API we spelled it out in separate roles (maybe an argument for doing the same?) but I don't have any strong opinions.